packages/design-system/src/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { forwardRef } from "react";
+import { useFocusRing } from "@react-aria/focus";
+import { useTextField } from "@react-aria/textfield";
+
+import { TextInputProps } from "./TextInput.types";
+import {
+  Description,
+  ErrorMessage,
+  Label,
+  MainContainer,
+  StyledTextInput,
+  TextInputContainer,
+} from "./TextInput.styles";
+import { useDOMRef } from "Hooks/useDomRef";
+
+const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+  (props, ref): JSX.Element => {
+    const { description, errorMessage, isRequired, label, labelPosition } =
+      props;
+    const textInputRef = useDOMRef(ref);
+    const { descriptionProps, errorMessageProps, inputProps, labelProps } =
+      useTextField(props, textInputRef);
+
+    return (
+      <MainContainer labelPosition={labelPosition}>
+        <Label {...labelProps}>
+          {label}
+          {isRequired && <span>*</span>}
+        </Label>
+        <TextInputContainer>
+          <StyledTextInput type={"text"} {...inputProps} ref={textInputRef} />
+          {description && (
+            <Description {...descriptionProps} style={{ fontSize: 12 }}>
```

Comment:
If we're doing these things now because we don't have a typography scale or the needed components yet, can we leave a TODO so that we know we need to come back to this?